### PR TITLE
APP-3385: Logs tab: Add Filtering options to sidebar

### PR DESCRIPTION
**Pull request by Oabraham1**
[Ticket](https://viam.atlassian.net/browse/APP-3385)

In this PR, I added an extra field to the `GetRobotPartLogsRequest` message called `levels`. The goal of `levels` is to add an extra filtering option based on the level of a particular log in the database when fetching the logs.

This PR has to be merged before [this pr](https://github.com/viamrobotics/app/pull/3530) can be closed out since the `request.level` filed needs to be provided in the `app/service/server.go` file.

### CODE REVIEW

**Comment by Oabraham1** [comment]
I was also thinking I could just make the `filter` field an array of strings and use that for the filtering @michaellee1019

**Comment by michaellee1019** [comment]
The changes LGTM. Make sure to implement in a backwards compatible way on the backend: If an empty list is passed in then we should default to returning all levels.

I'm checking to see if we need to scope this.

**Review by stuqdog — APPROVED** [review]
Looks reasonable to me, agree with @njooma re: SDK perspective

File: proto/viam/app/v1/app.proto [hunk 1]
```diff
@@ -603,10 +603,20 @@ message GetRobotPartResponse {
 }
 
 message GetRobotPartLogsRequest {
+  enum LogLevel {
```
Comment (RoxyFarhad) [inline proto/viam/app/v1/app.proto:606]: can you move this out of the message for LogsRequest? It may be useful for other requests

**Review by RoxyFarhad — APPROVED** [review]
approving under assumption that you move the enum out of the message

File: proto/viam/app/v1/app.proto (hunk 1 above)
Comment (Oabraham1) [reply proto/viam/app/v1/app.proto]: Done

**Comment by DTCurrie** [comment]
> The changes LGTM. Make sure to implement in a backwards compatible way on the backend: If an empty list is passed in then we should default to returning all levels.
> 
> I'm checking to see if we need to scope this.

Can we document this is the expected behavior in some comments?

**Comment by Oabraham1** [comment]
> > The changes LGTM. Make sure to implement in a backwards compatible way on the backend: If an empty list is passed in then we should default to returning all levels.
> > I'm checking to see if we need to scope this.
> 
> Can we document this is the expected behavior in some comments?

I could add it to the ticket description or comments if that's what you mean?

**Comment by DTCurrie** [comment]
> > > The changes LGTM. Make sure to implement in a backwards compatible way on the backend: If an empty list is passed in then we should default to returning all levels.
> > > I'm checking to see if we need to scope this.
> > 
> > 
> > Can we document this is the expected behavior in some comments?
> 
> I could add it to the ticket description or comments if that's what you mean?

Actually, this change is going to require a scope doc. We will have to let the Docs team know and clarify exactly how we expect this API to behave. We expect this API to be used externally, so we have to clearly communicate any changes we are making.

**Comment by Oabraham1** [comment]
[Change scope doc](https://docs.google.com/document/d/1YLAixH7DS7PvNZUfsxx9DvqqAgH6W80_7xtGVQLg9aM/edit?usp=sharing) was approved

File: proto/viam/app/v1/app.proto [hunk 2]
```diff
@@ -604,9 +604,10 @@ message GetRobotPartResponse {
 
 message GetRobotPartLogsRequest {
   string id = 1;
-  bool errors_only = 2;
+  bool errors_only = 2 [deprecated = true];
```
Comment (DTCurrie) [inline proto/viam/app/v1/app.proto:607]: Should we include a comment pointing to the ticket to remove this down the road?

**Review by DTCurrie — APPROVED** [review]
✅

File: proto/viam/app/v1/app.proto (hunk 2 above)
Comment (Oabraham1) [reply proto/viam/app/v1/app.proto:609]: Added the comment

**Pull request merged**